test/unit/dao/OptimisticTimelock.test.ts

```diff
@@ -0,0 +1,63 @@
+import { expectRevert, getAddresses, getCore } from '@test/helpers';
+import { expect } from 'chai';
+import hre, { ethers } from 'hardhat';
+import { Signer } from 'ethers';
+
+const toBN = ethers.BigNumber.from;
+
+const impersonatedSigners: { [key: string]: Signer } = {};
+
+describe('TimelockedDelegator', function () {
+  let userAddress;
+  let governorAddress;
+
+  before(async () => {
+    const addresses = await getAddresses();
+
+    // add any addresses you want to impersonate here
+    const impersonatedAddresses = [
+      addresses.userAddress,
+      addresses.secondUserAddress,
+      addresses.pcvControllerAddress,
+      addresses.governorAddress
+    ];
+
+    for (const address of impersonatedAddresses) {
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [address]
+      });
+
+      impersonatedSigners[address] = await ethers.getSigner(address);
+    }
+  });
+
+  beforeEach(async function () {
+    const addresses = await getAddresses();
+
+    userAddress = addresses.userAddress;
+    governorAddress = addresses.governorAddress;
+
+    this.core = await getCore();
+
+    this.delay = toBN(1000);
+    this.timelock = await (
+      await ethers.getContractFactory('OptimisticTimelock')
+    ).deploy(this.core.address, this.delay, [], []);
+  });
+
+  describe('Become Admin', async function () {
```

Comment:
I would still probably add an integration test to check you could do something basic like send funds to this timelock and then send them elsewhere, but down to you

Reply:
https://github.com/volt-protocol/volt-protocol-core/pull/80/files#diff-8ce5ef25e4a25a41b3fa57372221b2f35190ea61aedc05943735332aaee1f41f